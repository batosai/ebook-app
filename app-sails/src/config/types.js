import { PropTypes as T } from 'react';

export const Library = T.shape({
  id: T.string.isRequired,
  name: T.string.isRequired,
});
