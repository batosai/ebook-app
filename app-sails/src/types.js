export const BOOKS_REQUEST = 'books:request';
export const BOOKS_SUCCESS = 'books:success';
export const BOOKS_FAILURE = 'books:failure';

export const BOOK_REQUEST = 'book:request';
export const BOOK_SUCCESS = 'book:success';
export const BOOK_EDIT_REQUEST = 'book:edit:request';
export const BOOK_EDIT_SUCCESS = 'book:edit:success';
export const BOOK_DELETE_REQUEST = 'book:delete:request';
export const BOOK_DELETE_SUCCESS = 'book:delete:success';

export const ASIDE_TOGGLE = 'aside:toggle';

export const COLLECTIONS_REQUEST = 'collections:request';
export const COLLECTIONS_SUCCESS = 'collections:success';
export const COLLECTIONS_FAILURE = 'collections:failure';
export const COLLECTIONS_FILTER = 'collections:filter';

export const COLLECTION_ADD_REQUEST = 'collection:add:request';
export const COLLECTION_EDIT_REQUEST = 'collection:edit:request';
export const COLLECTION_DELETE_REQUEST = 'collection:delete:request';

export const LIBRARIES_REQUEST = 'libraries:request';
export const LIBRARIES_SUCCESS = 'libraries:success';
export const LIBRARIES_FAILURE = 'libraries:failure';

export const LIBRARY_ADD_REQUEST = 'library:add:request';
export const LIBRARY_EDIT_REQUEST = 'library:edit:request';
export const LIBRARY_DELETE_REQUEST = 'library:delete:request';
