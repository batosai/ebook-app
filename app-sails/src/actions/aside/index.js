// import * as types from '../../types';
//
// export const toggleAsideAction = () => ({
//   type: types.ASIDE_TOGGLE
// });

import { createTypes, createActions } from '../../utils';

export const types = createTypes('ASIDE:TOGGLE');
export const actions = createActions(types);

export const dispatch = () => dispatch => {
  dispatch(actions.request());
};
