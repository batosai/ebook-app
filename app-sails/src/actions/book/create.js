import { createTypes, createActions, request } from '../../utils';

export const types = createTypes("BOOK:CREATE")
export const actions = createActions(types)

export const dispatch = data => dispatch => {
  dispatch(actions.request());

  request({ method:'POST', url: '/books', data })
    .then(({body, response}) => dispatch(actions.success(body)));
}
