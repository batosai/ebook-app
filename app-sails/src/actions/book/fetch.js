import { createTypes, createActions, request } from '../../utils';

export const types = createTypes("BOOK:FETCH")
export const actions = createActions(types)

export const dispatch = data => dispatch => {
  dispatch(actions.request());

  request({ method:'GET', url: '/books', data })
    .then(({body, response}) => dispatch(actions.success(body)));
}


// import * as types from '../types';
//
// export const booksRequestAction = id => ({
//   type: types.BOOKS_REQUEST,
//   emit:true,
//   method:'GET',
//   url: '/books',
//   data: { collection: id }
// });
