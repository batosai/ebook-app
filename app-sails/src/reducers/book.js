import { book } from '../actions';

const initialState = { /* {id, img, title, collection_id, author, editor, formats, number_pages, number_volumes, year, read, keyword, description} */ };

const reducers = (state=initialState, {type, payload}) => {
  switch (type) {
    case book.fetch.types.SUCCESS:
      return payload.body;
    default:
      return state;
  }
}

export default reducers;
