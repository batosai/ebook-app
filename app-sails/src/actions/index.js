// import { toggleAsideAction } from './aside';
// export { toggleAsideAction };

// import { booksRequestAction } from './books';
// export { booksRequestAction };
//
// import { bookRequestAction, bookEditRequestAction, bookDeleteRequestAction } from './book';
// export { bookRequestAction, bookEditRequestAction, bookDeleteRequestAction };
//
// import { collectionsRequestAction, collectionByLibraryAction, collectionAddRequestAction, collectionEditRequestAction, collectionDeleteRequestAction } from './collections';
// export { collectionsRequestAction, collectionByLibraryAction, collectionAddRequestAction, collectionEditRequestAction, collectionDeleteRequestAction };
//
// import { librariesRequestAction, libraryAddRequestAction, libraryEditRequestAction, libraryDeleteRequestAction } from './libraries';
// export { librariesRequestAction, libraryAddRequestAction, libraryEditRequestAction, libraryDeleteRequestAction };


import * as aside from './aside';
import * as book from './book';
import * as collection from './collection';
import * as library from './library';
export {
  aside,
  book,
  collection,
  library,
}
