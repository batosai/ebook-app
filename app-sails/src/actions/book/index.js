import * as create from './create';
import * as update from './update';
import * as remove from './remove';
import * as fetch from './fetch';
import * as upload from './upload';
import * as illustration from './illustration';

const dispatch = (dispatch, event) => {
  switch (event.verb) {
    case 'created':
      dispatch(create.actions.success(event.data));
      break;
    case 'updated':
      dispatch(update.actions.success(event.data));
      break;
    case 'destroyed':
      dispatch(remove.actions.success(event.id));
      break;
    default:
      console.warn(
        'Unrecognized socket event (`%s`) from server:',
        event.verb,
        event,
      );
  }
};

export { create, update, remove, fetch, dispatch, illustration, upload };
