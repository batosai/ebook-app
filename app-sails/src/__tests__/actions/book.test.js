import * as actions from '../../actions/book';
import * as types from '../../types';

describe('Action', () => {
  it('to request book', () => {
    const expectedAction = {
      type: types.BOOK_REQUEST,
      emit: true,
      payload: { id: 1 },
    };

    expect(actions.bookRequestAction(1)).toEqual(expectedAction);
    expect(actions.bookRequestAction(1)).toMatchSnapshot();
  });

  it('to edit book', () => {
    const expectedAction = {
      type: types.BOOK_EDIT_REQUEST,
      emit: true,
      payload: {
        book: {
          title: 'my title',
          read: false,
          author: 'jeremy',
          editor: 'glÃ©nat',
          number_pages: 22,
          number_volumes: 4,
          year: 2010,
          collection_id: 2,
          description: '',
        },
      },
    };

    expect(actions.bookEditRequestAction(expectedAction.payload.book)).toEqual(
      expectedAction,
    );
    expect(
      actions.bookEditRequestAction(expectedAction.payload.book),
    ).toMatchSnapshot();
  });

  it('to delete book', () => {
    const expectedAction = {
      type: types.BOOK_DELETE_REQUEST,
      emit: true,
      payload: { id: 1 },
    };

    expect(actions.bookDeleteRequestAction(1)).toEqual(expectedAction);
    expect(actions.bookDeleteRequestAction(1)).toMatchSnapshot();
  });
});
