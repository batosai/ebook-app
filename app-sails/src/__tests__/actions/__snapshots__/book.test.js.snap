exports[`Action to delete book 1`] = `
Object {
  "emit": true,
  "payload": Object {
    "id": 1,
  },
  "type": "book:delete:request",
}
`;

exports[`Action to edit book 1`] = `
Object {
  "emit": true,
  "payload": Object {
    "book": Object {
      "author": "jeremy",
      "collection_id": 2,
      "description": "",
      "editor": "gl√©nat",
      "number_pages": 22,
      "number_volumes": 4,
      "read": false,
      "title": "my title",
      "year": 2010,
    },
  },
  "type": "book:edit:request",
}
`;

exports[`Action to request book 1`] = `
Object {
  "emit": true,
  "payload": Object {
    "id": 1,
  },
  "type": "book:request",
}
`;
